cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillPreparer.java

```diff
@@ -1106,23 +1108,31 @@ private V1Container createContainer(String name, String containerImage, String i
    * Creates a {@link V1ResourceRequirements} based on the given {@link ResourceSpecification}. If the namespace has a
    * resource quota, the objects must also specify resource limits.
    */
-  private V1ResourceRequirements createResourceRequirements(ResourceSpecification resourceSpec) {
+  private V1ResourceRequirements createResourceRequirements(Type resourceType, ResourceSpecification resourceSpec) {
     Map<String, String> cConf = masterEnvContext.getConfigurations();
     float cpuMultiplier = Float.parseFloat(cConf.getOrDefault(CPU_MULTIPLIER, DEFAULT_MULTIPLIER));
     float memoryMultiplier = Float.parseFloat(cConf.getOrDefault(MEMORY_MULTIPLIER, DEFAULT_MULTIPLIER));
+
+    if (resourceType == V1Job.class) {
+      // For job type use program multipliers.
+      cpuMultiplier = Float.parseFloat(cConf.getOrDefault(PROGRAM_CPU_MULTIPLIER, DEFAULT_PROGRAM_CPU_MULTIPLIER));
```

Comment:
do we need to do any validation on this in case it's over 1? Or does Kubernetes handle the case when the limit is lower than the request?

Reply:
on kubernetes side its invalid to specify limits lower than requests, but as a safe guard its good to add that validation here. Will make that change.

Reply:
done